src/plugin/transform.ts

```diff
@@ -35,78 +27,118 @@ type TransformOptions = {
-
-            const token = state.push(TOKEN_TYPE, TAG, 0);
-            const htmlBlockId = generateHtmlBlockId();
-            const className = [BLOCK_NAME, containerClasses].filter(Boolean).join(' ');
-
-            token.block = true;
-
-            token.attrSet(TokenAttr.class, className);
-            token.attrPush([TokenAttr.dataId, htmlBlockId]);
-            token.attrPush([TokenAttr.dataKey, BLOCK_NAME]);
-            token.attrPush([TokenAttr.frameborder, '0']);
-            token.attrPush([TokenAttr.id, htmlBlockId]);
-            token.attrPush([TokenAttr.style, 'width:100%']);
-            token.attrPush([TokenAttr.srcdoc, content]);
-
-            // TODO: use collect
-            env.meta = env.meta || {};
-            env.meta.script = env.meta.script || [];
-            env.meta.style = env.meta.style || [];
-            if (!env.meta.script.includes(runtimeJsPath)) {
-                env.meta.script.push(runtimeJsPath);
-            }
-
-            if (bundle) {
-                copyRuntimeFiles({runtimeJsPath, output}, env.bundled);
-            }
-
-            return true;
-        };
-
-        // the directives plugin must be enabled
-        md.use(directivePlugin);
-
-        const mdDir = md as MarkdownItWithDirectives;
-        mdDir.blockDirectives['html'] = plugin;
-        mdDir.renderer.rules[TOKEN_TYPE] = (tokens, idx, _opts, _env, self) => {
-            const token = tokens[idx];
-
-            const base = `<base target="${baseTarget}">`;
-            let content = base + token.attrGet(TokenAttr.srcdoc) ?? '';
-
-            if (styles) {
-                const stylesContent =
-                    typeof styles === 'string'
-                        ? `<link rel="stylesheet" href="${styles}" />`
-                        : `<style>${getStyles(styles)}</style>`;
-                content = stylesContent + content;
-            }
-
-            const resultContent = sanitize ? sanitize(content) : content;
-            token.attrSet(TokenAttr.srcdoc, resultContent);
-
-            return `<${token.tag}${self.renderAttrs(token)}></${token.tag}>`;
-        };
+        registerTransform(md, {runtimeJsPath, bundle, output, updateTokens: true});
+
+        (md as MarkdownItWithDirectives).renderer.rules[ISOLATED_TOKEN_TYPE] =
+            makeIsolatedModeEmbedRenderRule({
+                containerClassNames: containerClasses,
+                baseTarget,
+                isolatedSandboxHost,
+                embedContentTransformFn: (raw) => concatStylesIncludeDirectives(raw, styles),
+            });
+
+        (md as MarkdownItWithDirectives).renderer.rules[SHADOW_TOKEN_TYPE] =
+            makeShadowModeEmbedRenderRule({
+                containerClassNames: containerClasses,
+                embedContentTransformFn: (raw) => {
+                    const withStyles = concatStylesIncludeDirectives(raw, styles);
+
+                    // Note that sanitization is only enabled for `shadow` embedding mode,
```

Comment:
I suggest adding the sanitizer option for the iframe mode too, just do not set it by default